web/src/components/ErrorScreen.js

```diff
@@ -1,122 +1,138 @@
 import React from 'react';
 import Guild from './Guild';
 import {UserAvatar} from './DiscordIcon';
+import {CSSTransitionGroup} from 'react-transition-group';
 
-export default class ErrorScreen extends React.Component {
-  render() {
-    const props = this.makeInner();
-    if (!props) return null;
+export default class ErrorScreenWrapper extends React.Component {
+  renderNoAuth() {
+    return (
+      <ErrorScreen header="Huh?!">
+        <section>
+          <p>Well, this is embarrassing. I don't know how you got here!</p>
+          <p>This isn't normal at all, so I'm not really sure how to proceed...</p>
+          <p>Maybe try refreshing? Let's hope for the best!</p>
+        </section>
+      </ErrorScreen>
+    );
+  }
 
+  renderDisconnected() {
     return (
-      <div className="front flex-vertical flex-spacer">
-        <div className="application-icon">
-          <div className="application-icon-inner" />
-        </div>
-        <div className="front-inner">
-          <header>
-            <h1>{props.header}</h1>
-          </header>
-          <div className="scroller-wrap">
-            <div className="details scroller">{props.body}</div>
-          </div>
-          <footer>{props.footer}</footer>
+      <ErrorScreen header="Disconnected!">
+        <section>
+          <p>Sorry! I might have gone to sleep for the night!</p>
+          <p>If you know I'm supposed to be awake, try giving this page a refresh!</p>
+        </section>
+      </ErrorScreen>
+    );
+  }
+
+  renderNoAccess(data) {
+    return (
+      <ErrorScreen
+        header="Access denied!"
+        footer={
+          <form method="POST" action="logout">

```

Comment:
nit: i'd prefer putting this in a variable first

Reply:
What do you mean? The whole return value? Or just the footer stuff?

Reply:
just the footer